### DISCUSSION TITLE
[2021/11/16] Frontend Night

### DISCUSSION THREAD

**Discussion by jjenko** [wantedly/frontend-night#178] [Archive]
## Policy

- 資料の準備をがんばらない
- 参加・途中退席は自由
- スキップしない

話したい・聞きたいネタを書いてく ✏️

ハッシュタグ [#frontend_night](https://twitter.com/hashtag/frontend_night?src=hash)

Prev: https://github.com/wantedly/frontend-night/issues/165 

 _Created from https://github.com/wantedly/frontend-night/issues/65 by [issue-creator](https://github.com/rerost/issue-creator)_

**Comment by kobayang** [comment]
## Custom ESLint rule

発表練習ついでにまとめたことを話す

**Comment by chloe463** [comment]
## [The Large, Small, and Dynamic Viewports](https://www.bram.us/2021/07/08/the-large-small-and-dynamic-viewports/?utm_source=pocket_mylist)

view port 関連で新しい単位ができるらしい

**Reply by izumin5210** [reply]
まだ真っ赤 https://caniuse.com/viewport-unit-variants

**Comment by kobayang** [comment]
## React Location

https://react-location.tanstack.com/overview
https://twitter.com/tannerlinsley/status/1460274803139510278
https://github.com/tannerlinsley/react-location

新しく出たReactのRouterライブラリ。

React Router と比較すると、より高機能（Prefetching, Error, Data Caching...）をサポートしている。
https://react-location.tanstack.com/comparison

Route 定義は react-router に似ている（もともと React Router v6 の wrapper からスタートしている）ようにみえるので、 react-router → react-location への移行も比較的簡単そう（？）

**Reply by kobayang** [reply]
react-location の loader と react-query を組み合わせる exmaple
https://github.com/tannerlinsley/react-location/blob/main/examples/with-react-query/src/index.tsx

loader で非同期ロジックが書けるので、cache があるAPIクライアント用のライブラリと相性が良さそう。
Prefetch がすごく簡単に実現できるようになる。

c.f. https://twitter.com/takepepe/status/1460283914837057538

**Comment by chloe463** [comment]
## CRA はメンテナンスモードらしい
https://twitter.com/oukayuka/status/1460079431460929536?s=20

> but overall, starting with 2.0, it's mostly in maintenance mode and does not strive to be the best tool for production React apps.

2.0 からは、ほとんどメンテナンスモードになっていて、production react apps を作るためのベストツールではない

> So if you care about delivering the best user experience, I don't think CRA is the right tool in the first place.

最高のUXを届けるためのツールを考えているなら、CRAは第一候補ではない

> CRA still works great for the getting started use case, but you shouldn't see it as the best React app setup

CRAは「とりあえずはじめよう」くらいのユースケースには向いているが、React app のセットアップのための最善策とは考えないほうがよい

**Comment by chloe463** [comment]
## [So You Think You Can Build A Dropdown? - Pedro Duarte - (Next.js Conf 2021)](https://www.youtube.com/watch?v=pcMYcjtWwVI)

- アクセシビリティを意識した Dropdown 作れてますか？
	- キーボード操作や、voice over 対応してる？
- 見た目だけ作るのは簡単。機能をたくさんつけようとするとものすごく難しい。
- フォーカスしたいのはプロダクトの方であって、汎用の Dropdown コンポーネントを作ることではない。
- そういうのはこういうライブラリにまかせて、君たちはプロダクトコードを書くことに集中して。

**Reply by kobayang** [reply]
- downshift: https://github.com/downshift-js/downshift
- https://react-spectrum.adobe.com/react-spectrum/Menu.html

**Reply by kobayang** [reply]
この人が出しているライブラリ
https://www.radix-ui.com/

**Comment by chloe463** [comment]
## [Interactive stories (beta)](https://storybook.js.org/blog/interactive-stories-beta/)

Storybook 上で操作を再現できる機能の beta が入った模様
